src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/CreateTake.kt

```diff
@@ -0,0 +1,134 @@
+        val verseFormat = if (chunkCount > 99) "%03d" else "%02d"
+
+        // Format each piece of the filename
+        val languageSlug = project.resourceContainer?.language?.slug ?: ""
+        val rcSlug = project.resourceContainer?.identifier ?: ""
+        val bookNumber = "%02d".format(
+                // Handle book number offset (only for Bibles)
+                if (project.resourceContainer?.subject?.toLowerCase() == "bible"
+                        && project.sort > 39) project.sort + 1 else project.sort
+        )
+        val bookSlug = project.slug
+        val chapterNumber = chapterFormat.format(chapter.titleKey.toInt())
+        val verseNumber = if (chunk.start == chunk.end) {
+            verseFormat.format(chunk.start)
+        } else {
+            "$verseFormat-$verseFormat".format(chunk.start, chunk.end)
+        }
+        val takeNumber = "%02d".format(number)
+
+        // Compile the complete filename
+        return listOf(
+                languageSlug,
+                rcSlug,
+                "b$bookNumber",
+                bookSlug,
+                "c$chapterNumber",
+                "v$verseNumber",
+                "t$takeNumber"
+        ).joinToString("_", postfix = ".wav")
+    }
+
+    fun create(chunk: Chunk, project: Collection, chapter: Collection): Single<Take> = Single
+            .zip(
+                    getHighestTakeNumber(chunk),
+                    getNumberOfSubcollections(project),
+                    getNumberOfChunks(chapter),
+                    Function3 { highest, chapterCount, verseCount ->
+                        val filename = generateTakeFilename(
+                                project,
+                                chapter,
+                                chunk,
+                                highest + 1,
+                                chapterCount,
+                                verseCount
+                        )
+
+                        // Create a file for this take
+                        val chapterFormat = if (chapterCount > 99) "%03d" else "%02d"
+
+                        val takeFile = directoryProvider
+                                .getProjectAudioDirectory(project, chapterFormat.format(chapter.titleKey.toInt()))
+                                .resolve(File(filename))
+
+                        val newTake = Take(
+                                takeFile.name,
+                                takeFile,
+                                highest + 1,
+                                LocalDate.now(),
+                                false,
+                                listOf() // No markers
+                        )
+
+                        // Create an empty WAV file
+                        waveFileCreator.createEmpty(newTake.path)
+                        return@Function3 newTake
+                    }
+            )
+
+    fun saveNew(take: Take, chunk: Chunk): Completable = takeRepository
```

Comment:
Chunk, Take

Is there ever a situation where tracking the file in the database is not part of creating a take?

Reply:
My intention was to separate these responsibilities into functions like I separated the responsibilities of getting the take count, getting the max take number, or generating the filename.

Reply:
yeah; I don't like the idea of having to remember to commit the file to the database especially if that happens every time

I'm okay with them being private, I'm also okay with adding the code from saveNew to create, keeping create private, and making this the record use case